### PR TITLE
Feature/delete user profile backend

### PR DESCRIPTION
**Pull request by aussie-m-mike**

### CODE REVIEW

File: api/app/Actions/User/DeleteUserAction.php [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            $this->imageUploader->remove($user->avatar);
+        }
+
+        if ($user->branding_logo) {
+            $this->imageUploader->remove($user->branding_logo);
+        }
+
+        $user->delete();
+
+        return new DeleteUserResponse($user);
```
Comment (aLLeXUs) [inline api/app/Actions/User/DeleteUserAction.php:42]: I am not sure we need to return the deleted user.

File: api/app/Entity/User.php [hunk 2]
```diff
@@ -78,4 +79,14 @@ public function eventTypes()
...
+
+    public function getAvatarUrl()
+    {
+        return $this->avatar ? Storage::disk()->url($this->avatar) : null;
+    }
+
+    public function getBrandingLogoUrl()
+    {
+        return $this->branding_logo ? Storage::disk()->url($this->branding_logo) : null;
+    }
```
Comment (aLLeXUs) [inline api/app/Entity/User.php:91]: Return types missing.

File: api/app/Actions/User/DeleteUserResponse.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\User;
+
+use App\Entity\User;
+
+final class DeleteUserResponse
```
Comment (aLLeXUs) [inline api/app/Actions/User/DeleteUserResponse.php:9]: The response most likely not needed.

**Comment by aussie-m-mike** [comment]
Fixed

File: api/app/Http/Controllers/Api/UserController.php [hunk 4]
```diff
@@ -41,8 +46,10 @@ public function store(UserRequest $request): JsonResponse
         return $this->successResponse($this->userArrayPresenter->present($response->getProfile()));
     }
 
-    public function delete($id)
+    public function delete(): JsonResponse
     {
-        //
+        $this->deleteUserAction->execute(new DeleteUserRequest(Auth::id()));
+
+        return $this->successResponse([]);
```
Comment (aLLeXUs) [inline api/app/Http/Controllers/Api/UserController.php:53]: `return $this->emptyResponse();`

**Comment by aussie-m-mike** [comment]
done

**Pull request merged**